robottelo/hosts.py

```diff
@@ -1101,17 +1112,9 @@ def configure_insights_client(
         # Ensure insights-client rpm is installed
         if self.execute('yum install -y insights-client').status != 0:
             raise ContentHostError('Unable to install insights-client rpm')
-        # attempt to register host
-        if register:
-            if not activation_key:
-                activation_key = satellite.api.ActivationKey(
-                    content_view=org.default_content_view.id,
-                    environment=org.library.id,
-                    organization=org,
-                ).create()
-            self.register(
-                org, None, activation_key.name, satellite, setup_insights=register_insights
-            )
+        # Register client with insights
+        if self.execute('insights-client --register').status != 0:
+            raise ContentHostError('Unable to register client with insights')
```

Comment:
I think manual install and register for insight-client would be required only when register=False, so could this move in if-else?

Reply:
This code is weird. Basically, we register a host in order to create the custom repos needed for insights(these base OS repos are too big to enable and sync every test). Insights client does not register during this. Its only when we install packages and manually register at the end that insights gets set up.

Reply:
+1 to the workflow, it's dumb but this works.

Reply:
@ColeHiggins2 I don't understand why are we moving the registration part at the start of the `configure_insights_client`.  We probably should remove `yum install -y insights-client` part here as the `insights-client` should get install during host registration process, we don't need to do it manually.

Reply:
Since its general helper which install and insights during registration process when `register=True` ,  and when `register=False` we should do these setup manually

Reply:
It's probably better to get rid of this helper, it's very old and it's usage is now very confusing. We can keep the part to add custom repo on host and then do registration like we normally do using global registration command.

Reply:
@ColeHiggins2 I already suggested the same thing on your last PR https://github.com/SatelliteQE/robottelo/pull/16497#discussion_r1786745731, 
But if we consider getting rid of this helper itself, so I'd suggest using `repo_data` option in global registratio, with this option while registering we could pass custom repos and it creates foreman_register repo in /etc/yum.repos.d/. 
For 6.16 and stream, it does support creating multiple repos now, and it creates foreman_register1, foreman_register2 repo files in /etc/yum.repos.d/